src/plugins/intel_gpu/src/graph/include/custom_gpu_primitive_inst.h

```diff
@@ -50,6 +55,32 @@ class typed_primitive_inst<custom_gpu_primitive> : public typed_primitive_inst_b
 
 public:
     typed_primitive_inst(network& network, custom_gpu_primitive_node const& node);
+
+private:
+    static void update_output_shape(const kernel_impl_params& impl_param, layout& output_layout) {
+        bool is_dynamic_input = false;
+        const auto inp_sz = impl_param.get_input_layout_size();
+        for (size_t i = 0; i < inp_sz; i++) {
+            if (impl_param.get_input_layout(i).is_dynamic()) {
+                is_dynamic_input = true;
+                break;
+            }
+        }
+
+        if (!is_dynamic_input && output_layout.is_dynamic()) {
```

Comment:
nit:
It would be good to add comment explaining this, something like:
_Execute the op's shape inference only for dynamic node when input shapes have already been calculated; otherwise, keep the original output layout unchanged (it will be either static for static model or have dynamic shape in case of dynamic flow)_